applications/launchpad/backend/src/api/wallet_api.rs

```
@@ -64,6 +65,13 @@ pub async fn wallet_balance() -> Result<WalletBalance, String> {
 
 #[tauri::command]
 pub async fn wallet_events(app: AppHandle<Wry>) -> Result<(), String> {
+    loop {
+        if "running" == status(ImageType::Wallet).await.to_lowercase() {
+            break;
+        } else {
+            thread::sleep(Duration::from_secs(1))
+        }
+    }
```

This approach is brittle, possibly broken.

It only checks if the container is running; not if the wallet is responsive.

Let us fail-fast and defer the responsibility of waiting for the wallet to be running to the front-end. This command does not need to check for that. If the client connection fails, we can return an error and the FE can react accordingly.

---

I do agree the code should be changed according to code quality standards. The front end calls this method once. If I remove it then the front end should change the current behaviour (passive) and start listen actively.